.github/workflows/main.yml

```diff
@@ -0,0 +1,98 @@
+name: CI
+
+on: [ push, pull_request ]
+
+jobs:
+  lockfile-lint:
+    name: Lockfile lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - name: lint lock file
+        run: npx lockfile-lint --path package-lock.json --allowed-hosts npm --validate-https
+
+  quality-checks:
+    name: Quality checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - name: install dependencies
+        run: npm ci
+      - name: types check
+        run: npm run types-check
+      - name: lint check
+        run: npm run lint
+
+  test:
+    strategy:
+      matrix:
+        platform: [ ubuntu-latest ]
+        node: [ '10', '12', '14', '16' ]
+    needs: [ 'lockfile-lint', 'quality-checks' ]
+    name: Node v.${{ matrix.node }} Tests (${{ matrix.platform }})
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ matrix.node }}-node-${{ hashFiles('**/package-lock.json') }}
+      - name: install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: npm ci
+      - name: unit tests
+        run: npm run unit-tests
+      - name: integration tests
+        run: npx npm run integration-tests
+      - name: coveralls
+        uses: coverallsapp/github-action@master
+        if: ${{ matrix.node == '14' }}
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+  prom-client-test:
+    strategy:
+      matrix:
+        node: [ '12', '14' ]
+        prom-client: [ '12' ]
+    needs: [ 'lockfile-lint', 'quality-checks' ]
+    name: Prom Client v.${{ matrix.prom-client }} Tests (node ${{ matrix.node }})
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - name: install dependencies
+        run: npm ci
+      - name: install prom client
+        run: npm i prom-client@${{ matrix.prom-client }}
+      - name: run tests
+        run: npm test
+
+  release:
+    name: release
```

Comment:
you need to grant the release job the required permissions as you can see here https://github.com/PayU/openapi-validator-middleware/blob/b37469d78236ba582de930af136074a7ba5f5950/.github/workflows/main.yml#L46

Reply:
added.
@kobik I saw also some guy [adding '0.0.0-development'](https://youtu.be/rCXq86FOlzQ?t=567) to the package.json `version` in order to prevent people from touching it. should we do it too?

Reply:
it's ok to set it to 0.0.0

Reply:
set to `0.0.0`